### PR TITLE
Add keypoint-detection task to Hub

### PR DESCRIPTION
**Pull request by merveenoyan**
I have opened PR to keypoint detection because we already have around 30 keypoint detection models on Hub (currently opening PRs to change pipeline tag), one model in transformers, and this doesn't fall under any other task.

Screenshot 2024-08-27 at 15 08 26

### CODE REVIEW

**Comment by merveenoyan** [comment]
also cc @NielsRogge

File: packages/tasks/src/tasks/index.ts [hunk 1]
```diff
@@ -131,6 +131,7 @@ export const TASKS_MODEL_LIBRARIES: Record<PipelineType, ModelLibraryKey[]> = {
 	"image-to-image": ["diffusers", "transformers", "transformers.js"],
 	"image-to-text": ["transformers", "transformers.js"],
 	"image-to-video": ["diffusers"],
+	"keypoint-detection": ["transformers"],
```
Comment (NielsRogge) [inline packages/tasks/src/tasks/index.ts:134]: What does this mean? Cause there's no keypoint detection pipeline in Transformers yet

File: packages/tasks/src/pipelines.ts [hunk 2]
```diff
@@ -658,6 +658,12 @@ export const PIPELINE_DATA = {
 		color: "blue",
 		hideInDatasets: true,
 	},
+	"keypoint-detection": {
+		name: "Keypoint Detection",
+		modality: "cv",
+		color: "red",
+		hideInDatasets: true,
+	},
```
Comment (pcuenca) [inline packages/tasks/src/pipelines.ts:666]: Should it be a subtask of `object-detection` instead? There's already a `face-detection` subtask, for example.

File: packages/tasks/src/tasks/index.ts (hunk 1 above)
Comment (pcuenca) [reply packages/tasks/src/tasks/index.ts:134]: > we currently have SuperPoint and will soon have ViTPose

I think we should tag those models then before the PR is merged

File: packages/tasks/src/pipelines.ts (hunk 2 above)
Comment (merveenoyan) [reply packages/tasks/src/pipelines.ts:666]: I think it's not. object detection is where you detect bounding boxes, but in this one outputs look like these: 
keypoint-detection-output
Screenshot 2024-08-29 at 14 01 01

File: packages/tasks/src/tasks/index.ts (hunk 1 above)
Comment (merveenoyan) [reply packages/tasks/src/tasks/index.ts:134]: I have opened PRs to 30-40 models and some are merged already I think, see

also Sapiens model will have a lot of keypoint models soon (it was one unified model repo, I sent authors a script to automate separation)

Screenshot 2024-08-29 at 14 02 42

**Review by osanseviero — COMMENTED** [review]
Looking nice!

File: packages/tasks/src/pipelines.ts [hunk 3]
```diff
@@ -658,6 +658,12 @@ export const PIPELINE_DATA = {
 		color: "blue",
 		hideInDatasets: true,
 	},
+	"keypoint-detection": {
+		name: "Keypoint Detection",
+		modality: "cv",
+		color: "red",
+		hideInDatasets: true,
```
[the remark this answers is not in the document]
Comment (merveenoyan) [reply packages/tasks/src/pipelines.ts:665]: actually there seems to be a lot, opening PRs now 
https://huggingface.co/datasets?sort=trending&search=mediapipe
https://huggingface.co/datasets?sort=trending&search=keypoint
https://huggingface.co/datasets?sort=trending&search=pose

[the remark this answers is not in the document]
Comment (merveenoyan) [reply packages/tasks/src/pipelines.ts:671]: just set to false

**Comment by merveenoyan** [comment]
lol I opened too many PRs Hub thinks I'm spamming

File: packages/tasks/src/tasks/index.ts (hunk 1 above)
Comment (pcuenca) [reply packages/tasks/src/tasks/index.ts:134]: Yes, I see 4 models now 👍

**Review by pcuenca — COMMENTED** [review]
LGTM with the contributions from #873

**Pull request merged**